### PR TITLE
feat(build): compile skill templates into SKILL.md under ADR-108 (A1)

### PR DESCRIPTION
**Pull request by rjmurillo**
### What

Adds the skill-template compile pipeline (milestone A1 of issue #5706): `build/scripts/skill_templates.py` renders `templates/skills/<name>.SKILL.md.tmpl` with `templates/skills/partials/*.mustache` into `.claude/skills/<name>/SKILL.md` through a restricted mustache grammar; `generate_skills.py` runs it before the copy loop and gains `--validate`; `build_all.py` allowlists exactly the template-owned targets in the REQ-003-010 guard and runs the compile in validate mode under `--check`; a `Skill Template Drift` gate joins `pre_pr_sequence.py`; `chevron==0.14.0` becomes a dev dependency. No template exists yet, so the tree renders nothing and every gate stays green by construction.

### Why

ADR-108 (PR #5724) amends ADR-107 property 1, REQ-003-010, and REQ-003 D4 for this one class so rule guidance can be authored once and bundled into a skill at the step it governs. This PR is the mechanism; the eight pilot templates land in A2 (TASK-027). Merge after #5724 so the guard exception has its authority on the trunk first.

## Specification References

| Type | Reference | Description |
|------|-----------|-------------|
| **Issue** | Refs #5706 | build(skills): compile SKILL.md from mustache templates so guidance is bundled into skills |
| **PR** | Refs #5724 | ADR-108 and the spec this PR implements (merged; main merged into this branch) |
| **Spec** | `.agents/specs/design/DESIGN-024-skill-guidance-excerpt-sync.md` | Template grammar, compile module, wiring, tests |
| **Spec** | `.agents/specs/tasks/TASK-025-pilot-excerpts-and-skill-edits.md` | A1 done definition |

These references do not close their linked items: issue #5706 closes with A2, which ships the pilot templates and the byte report.

## Changes

- `build/scripts/skill_templates.py`: `discover`, `discover_errors`, `owned_targets`, `check_grammar`, `render`, `compile_all`. Grammar: only `{{> slug}}` and `{{! }}` tags, checked recursively through every partial a template includes (cycle is an error); missing partial, partial without exactly one trailing newline, or a template name that fails the slug pattern or names no existing skill directory: exit 2; literal `{{` in the render: exit 1; NO-REGEN under a template-owned target (all three sentinel forms): file unchanged, WARN, exit 1. Reads and writes with explicit newline handling so CRLF templates round-trip byte for byte.
- `build/scripts/generate_skills.py`: compile before the copy loop, `--what-if` honored, new `--validate` that compares the render to the committed file and exits 1 naming drift without copying.
- `build/scripts/build_all.py`: `assert_no_claude_writes(allowed_paths=...)` subtracts the allowlist from created and modified paths only (a deleted template-owned target is still reported); the skills step threads `--check` into validate mode so nothing under `.claude/` is written during a check; a compile failure propagates as exit 2.
- `pyproject.toml`, `uv.lock`: `chevron==0.14.0` in both dev tables.
- `scripts/validation/checks_portability.py`, `scripts/validation/pre_pr_sequence.py`, `scripts/validation/pre_pr.py`, `tests/validation/test_pre_pr_sequence_registry.py`: the `Skill Template Drift` gate, its facade re-export, and its `EXPECTED_ORDER` entry.
- `tests/build_scripts/test_generate_skills_template_compile.py`, `tests/build_scripts/test_skill_templates_pilot_scope.py`, `tests/build_scripts/test_build_all.py`: positive, negative, config, and edge cases including subprocess exit codes; the pilot-scope test pins `discover()` on the real tree to an empty `PILOT` set with a non-vacuous negative control.
- `.agents/governance/GENERATOR-FILES.md`, `build/AGENTS.md`, `.claude/rules/generated-artifacts.md`, `.claude/rules/templates.md`, `templates/AGENTS.md`: the class row, the REQ-003-010 exception, and the regeneration command; the two rule mirrors under `.github/instructions/` and `src/copilot-cli/instructions/` regenerated.
- `scripts/validation/skill_md_portability_baseline.json`: regenerated after `templates.md` gained a `vendor-portability` marker for the `build/` paths it now names (a direct baseline bump was refused by the semantic-conflict guard, issue #4195).

## Engine decision (issue #5706 step 3)

`chevron` 0.14.0 was chosen over an in-tree partial expander: pure Python, MIT, zero dependencies, standard mustache syntax so templates read the same as the agent-template pipeline's future. The restricted grammar exists because chevron renders a missing partial and an unknown variable as empty text silently (probed on 2026-09-11), so the grammar and existence checks run before rendering and are the only thing that turns a typo into a failure. Cost accepted: an unmaintained dependency (last release 2021-01-02) of which one feature is used; REQ-025 defers a swap to an in-tree expander if that becomes a finding.

## Divergences from the issue and the design, all recorded in code

- Issue #5706 step 10 asks a NO-REGEN-marked `SKILL.md` to be skipped silently like the copy path. Here a sentinel under a template-owned file is WARN plus exit 1 (ADR-108 section 4 in #5724), because the sentinel exempts the file from its only drift gate; `build_all.py` reports that as its own exit 2.
- Issue #5657 (`references/` translation and multi-line prompt bodies) is untouched: the compile runs before the existing copy and translation, which are unchanged.
- The compile module is 577 lines and its test module 919 lines, both over the 500-line taste-lint advisory ceiling; kept whole so the grammar, render, and NO-REGEN branches are reviewable in one place. The taste count ratchet is unchanged at its baseline.

## Acceptance criteria

- [x] With no template on disk, `uv run python build/scripts/build_all.py --check` exits 0 and the `Skill Template Drift` gate passes.
- [x] A template naming a missing partial, carrying a forbidden tag, referencing a partial without one trailing newline, or named outside the slug pattern exits 2 and writes nothing.
- [x] A render containing literal `{{` exits 1 and writes nothing.
- [x] A NO-REGEN sentinel under a template-owned target (in-file HTML comment, `# NO-REGEN`, or `.noregen` sidecar) leaves the file unchanged, prints WARN, and exits 1 in write and validate mode.
- [x] `assert_no_claude_writes` passes an allowlisted template-owned write and still exits 2 for any other write or any deletion under `.claude/`.
- [x] `chevron==0.14.0` is in both dev tables and `tests/test_pyproject_dev_deps_parity.py` passes.
- [x] `Skill Template Drift` appears in `EXPECTED_ORDER` directly after `Generated Artifact Staleness`.

## Type of Change

- [ ] Bug fix (non-breaking change fixing an issue)
- [x] New feature (non-breaking change adding functionality)
- [ ] Breaking change (fix or feature causing existing functionality to change)
- [ ] Documentation update
- [ ] Infrastructure/CI change
- [ ] Refactoring (no functional changes)

## Reviewer Expectations

**Review kind / scrutiny / urgency**: full review, high scrutiny (touches the `.claude/` no-write guard), soft target; merge after #5724.

**Focus areas**: `build_all.py` allowlist subtraction and check-mode threading; `skill_templates.py` grammar recursion and exit-code taxonomy; the pilot-scope test as the widening control.

## Testing

Deliverables in this PR:

- [x] Tests added/updated
- [x] Manual testing completed
- [ ] No testing required (documentation only)

Run on the branch: `uv run pytest tests/build_scripts/test_generate_skills_template_compile.py tests/build_scripts/test_skill_templates_pilot_scope.py tests/build_scripts/test_build_all.py tests/build_scripts/test_generate_skills.py tests/validation/test_pre_pr_sequence_registry.py tests/test_pyproject_dev_deps_parity.py -q` gives `295 passed`; `uv run ruff check build/scripts scripts/validation tests/build_scripts tests/validation` is clean; `uv run python build/scripts/build_all.py --check` exits 0; `uv run python scripts/validation/pre_pr.py` reports no BLOCKING finding; mypy on the changed sources is clean. An independent opus code review returned one blocker (partials were not grammar-checked) and three minors; all fixed in the last four commits.

## Agent Review

Implemented by a sonnet implementer agent from the design and task documents in #5724; reviewed by an independent opus code-reviewer agent that received the diff only; gates re-run by the orchestrating session before this PR was opened.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_017M4qLAptnWDBp4aHuFUs6L

### CODE REVIEW

File: build/AGENTS.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Generators, mirror-sync helpers, and drift/parity gates for the agent, skill, ru
 ## Matters
 
 - `build/scripts/build_all.py --check` is the drift gate every PR must pass. Red means regenerate, never hand-edit the output tree.
-- REQ-003-010: no generator may write under `.claude/`. `build_all.py` asserts this after every run by snapshotting `.claude/` before and diffing after.
+- REQ-003-010: no generator may write under `.claude/`, except the template-owned skill files ADR-108 enumerates (`.claude/skills/<name>/SKILL.md` for each `templates/skills/<name>.SKILL.md.tmpl` present at run time). `build_all.py` asserts this after every run by snapshotting `.claude/` before and diffing after, passing that allowlist to `assert_no_claude_writes`.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply build/AGENTS.md:8]: Confirmed. ADR-108's frontmatter said implemented: false while build/AGENTS.md (and its Copilot-CLI mirrors) describe the .claude/ write-guard exception as active, enforced code. Milestone A1 is the "first implementation slice" the ADR's own Status section names as the enforcement trigger, and this PR is that slice, so the field was stale.

Fixed in daeda9b58: flipped implemented to true. Frontmatter-only change (status stays proposed, formal acceptance is a separate governance step), matching the precedent other ADRs use (for example ADR-031: status rejected, implemented true) and staying inside the adr-review-policy hook's implemented-field-only exemption.

File: build/scripts/build_all.py [hunk 2]
```diff
@@ -161,7 +187,9 @@ def _build_skills(repo_root: Path, config_path: Path, platform: str) -> Generato
         result.notices.append(f"{platform}: no artifacts.skills stanza; skipped")
         return result
 
-    rc = generate_skills.generate_skills(config_path, repo_root)
+    rc = generate_skills.generate_skills(config_path, repo_root, validate=check)
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply build/scripts/build_all.py:210]: Confirmed and reproduced. Only copilot-cli.yaml carries an artifacts.skills stanza; _build_skills returned before calling skill_templates.compile_all whenever a platform lacked one. A full, unfiltered build_all.py --check run still exercised the gate through copilot-cli, but a scoped run against vscode or visual-studio (the module docstring's own --platform example, applied to a stanza-less platform) reported exit 0 over a drifted or malformed .claude/skills/<name>/SKILL.md.

Fixed in 200646999: the compile/drift gate now runs regardless of the stanza; only the copy loop stays stanza-gated. Added a regression test (build_all._build_skills with no stanza, a drifted target, check=True) proving exit 2 and no write.

File: build/scripts/skill_template_grammar.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+import chevron  # type: ignore[import-untyped]
+
+_PARTIAL_EXT = "mustache"
+
+# Matches a triple-brace (raw) tag before the double-brace alternative, so
+# `{{{raw}}}` is captured whole instead of leaving a trailing `}` behind.
+_TAG_RE = re.compile(r"\{\{\{.*?\}\}\}|\{\{.*?\}\}", re.DOTALL)
+_SLUG = r"[a-z0-9]+(?:-[a-z0-9]+)*"
+_PARTIAL_TAG_RE = re.compile(rf"^\{{\{{>\s*({_SLUG})\s*\}}\}}$")
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply build/scripts/skill_template_grammar.py:107]: Confirmed. Reproduced: check_grammar("See {{> greet}} for details.\n") returned [] before this fix. _PARTIAL_TAG_RE only ever matched match.group(0), the extracted tag substring, never the line it sits on, so a syntactically valid partial tag embedded mid-sentence passed clean even though the module's own documented grammar requires it on its own line.

Fixed in b96f0400d: added _is_standalone, which checks the tag's own line (whitespace-stripped) equals the tag text exactly, and requires it for a partial tag to be accepted. Three existing tests used a mid-line or shared-line tag incidentally rather than as the behavior under test; their fixtures now put the tag on its own line so they keep testing what they say they test. Added three new tests: indented standalone stays clean, mid-line is now offending, two tags sharing one line are now offending.

File: build/scripts/skill_templates.py [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+            print(f"DRIFT: {target} differs from its template ({tmpl_path})", file=sys.stderr)
+            result.drifted.append(str(target))
+            result.exit_code = max(result.exit_code, 1)
+            continue
+
+        if what_if:
+            print(f"  Would write: {target}")
+            continue
+
+        target.write_text(rendered, encoding="utf-8", newline="\n")
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply build/scripts/skill_templates.py:425]: Confirmed, security finding holds. _name_validation_error checked whether .claude/skills/<name>/ itself was a symlink and whether it resolved inside .claude/skills/, but never checked the SKILL.md file inside it. A real, non-symlinked directory holding a symlinked SKILL.md passed both existing checks; compile_all's target.write_text(...) follows a symlink the same way any open() call does, so a render could escape .claude/skills/ through that file.

Fixed in 99a4876c5: added the file-symlink check to _name_validation_error. Added a negative test (test_skill_templates_symlink_security.py) proving discover/discover_errors/owned_targets exclude the name and that compile_all never writes through the link, with the external file's content asserted unchanged.

File: templates/AGENTS.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ Source of truth for Copilot CLI, VS Code, and Visual Studio agent mirrors only;
 - `tools` / `tools_vscode` / `tools_copilot` MUST stay block-style YAML lists. Inline arrays (`['a','b']`) break Copilot CLI on CRLF line endings (issue #893).
 - A `model:` pin appears in generated output only for the `haiku` tier or a validated ADR-080 `KEEP_PIN` manifest entry; `opus`, `sonnet`, or no `model_tier` all resolve to no pin.
 - Regenerating and committing output is not optional: an uncommitted generator run after a template edit is a protocol failure (`.claude/rules/templates.md` MUST-1/2).
+- `skills/<name>.SKILL.md.tmpl` and `skills/partials/*.mustache` are a second, unrelated template class (ADR-108): `build/scripts/skill_templates.py`, invoked from `build/scripts/generate_skills.py`, compiles them into `.claude/skills/<name>/SKILL.md`, the one exception to `.claude/` being off-limits to generators. Regenerate with `uv run python build/scripts/build_all.py`, not `generate_agents.py`.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply templates/AGENTS.md:13]: Confirmed. The Entry points list called build/generate_agents.py the only supported way to produce output from a template edit, contradicting the Matters section's own bullet (added earlier in this PR) naming build_all.py as the regeneration command for the skills/<name>.SKILL.md.tmpl class.

Fixed in d39ee25bc: scoped the claim to agents/<name>.shared.md edits and cross-referenced the skill-template exception.

File: tests/build_scripts/_skill_template_helpers.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    return config
+
+
+def run_cli(*args: str) -> subprocess.CompletedProcess[str]:
+    return subprocess.run(
+        [sys.executable, str(_GENERATE_SKILLS_PATH), *args],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply tests/build_scripts/_skill_template_helpers.py:77]: Confirmed, run_cli had no timeout: a hung generate_skills.py child process would wedge the test job instead of failing it.

Fixed in 41367b2bf: added timeout=60, matching the same-purpose timeout on the python3 subprocess in test_generate_pr_quality_prompts.py:586.

File: build/scripts/skill_templates.py [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+
+    resolved_root = skills_root.resolve()
+    resolved_skill = skill_dir.resolve()
+    if not resolved_skill.is_relative_to(resolved_root):
+        return (
+            f"{tmpl_path}: .claude/skills/{name}/ resolves to {resolved_skill}, "
+            f"outside {resolved_root}"
+        )
+
+    if (skill_dir / "SKILL.md").is_symlink():
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply build/scripts/skill_templates.py:243]: Fixed in 0095f8fee: _name_validation_error now rejects any .claude/skills root whose resolved path is outside repo_root.resolve(), before the per-skill check, so an intermediate .claude or .claude/skills symlink to an external tree is a config error (exit 2) and no target is computed. Regression test test_discover_excludes_every_skill_when_the_skills_root_is_a_symlink_outside_the_repo asserts the external SKILL.md stays unchanged.
